src/main.cpp

```diff
@@ -3,197 +3,101 @@
-    IGNITION = 0;
-    // send CAN msg to motor ID to stop
-    const char* data = "some stop data sequence";     
-    can.write(CANMessage(0,data,8));
-}
-
-
-// read throttle from can and send to motor 
-void checkThrottle(){
-   
-    // read dms and ignition
-    checkDMS();
-    
-    if(DMS && IGNITION){
-        // Read potentiometer from pedal and relay to motor
-        CANMessage msg;
-
-        //read analog in of pot (comes in as 0-1)
-        if(can.read(msg)){
-            
-            // Parse pot data and write to motor id
-            
-            const char* data = "01010100";
-            can.write(CANMessage(0,data,8));
-
-        }
-    }
-    else if(IGNITION){
-        TFT.locate(100,100);
-        TFT.printf("Waiting for hand..."); 
+int readDMS(){
+    float DMS_VAL = dms.read();
+    int DMS_FLAG =0;
+    if (DMS_VAL <= DMS_THRESH){
+        DMS_FLAG = 1;
     }
-    else{
-        TFT.locate(100,100);
-        TFT.printf("Waiting for ignition and DMS...");
+    else if (DMS_VAL > DMS_THRESH){
+        DMS_FLAG = 0;
     }
-    
-    // print data eg power, voltage, # of lap, elapsed time
-
+    return DMS_FLAG;
 }
 
-void setupAccessoriesInterrupts(){
-
-    debounce.start();
-
-    leftBlinker.rise(&controlAccessories);
-    rightBlinker.rise(&controlAccessories);
-    horn.rise(&controlAccessories);
-    hazards.rise(&controlAccessories);
-    wipers.rise(&controlAccessories);
+void MTR_task(void){
+    int DMS_FLAG    = readDMS();
+    int IGNITION    = ignition.read();
+    int BRAKE       = brake.read();
+    char ignitionVal    =    (char) (IGNITION);
+    char brakeVal       =    (char) (BRAKE<<1);
+    char DMSVal         =    (char) (DMS_FLAG<<2);
+    float throttleVal   = 0;
     
-    leftBlinker.fall(&controlAccessories);
-    rightBlinker.fall(&controlAccessories);
-    horn.fall(&controlAccessories);
-    hazards.fall(&controlAccessories);
-    wipers.fall(&controlAccessories);
-
-}
-
-// reads digital inputs and sends CAN msg to accessories upon any interrupts
-void controlAccessories(){
+    if (DMS_FLAG && IGNITION && !BRAKE){
+        throttleVal = throttle.read();
+    }   
     
-    // read digial input for interrupters
-    // Pack one string for can msg
-    // bit sequence: leftBlinker, rightBlinker, horn, hazards, wipers
-    string leftbVal = to_string(leftBlinker.read());
-    string rightbVal = to_string(rightBlinker.read());
-    string hornVal = to_string(horn.read());
-    string hazardVal = to_string(hazards.read());
-    string wiperVal = to_string(wipers.read());
-    
-    // "00110" = 6    
-
-    // some example sequence
-    const char* data =  "00110011";
-
-    // debounce and send data
-    if (debounce.read_ms()>200){
-        
-        // currently have arbitrary CAN ID of 1 and length 8
-        can.write(CANMessage(1,data,8));
+    else {
+        throttleVal = 0;
+    }
+    std::string tmp = std::to_string(throttleVal);
+    const char* throttleCANData = tmp.c_str();
```

Comment:
I'm curious what's going on here? Why is the throttle data being converted to a string and then to a single character? I thought we were going to scale the throttleVal so it's a number from 0-255 so it fits neatly into a byte, but maybe I'm misunderstanding

Reply:
Talisha mentioned that she wants the raw value we get from the potentiometer in a string for now. I think we can sort this out on Saturday in person

Reply:
Sending strings across CAN doesn't work very well because of the strictly-defined byte size. A floating point value like throttleVal will be anything from 0 to 1 with an unknown number of decimal points. So I strongly recommend we turn this to an integer and Talisha uses that directly on motor controller. Let's meet Saturday and discuss!